crates/client/db/src/rocksdb_snapshot.rs

```diff
@@ -0,0 +1,169 @@
+use librocksdb_sys as ffi;
+use rocksdb::{AsColumnFamilyRef, DBAccess, DBPinnableSlice, Error, ReadOptions};
+use std::sync::Arc;
+
+/// A copy of [`rocksdb::SnapshotWithThreadMode`] with an Arc<DB> instead of an &'_ DB reference
+pub struct SnapshotWithDBArc<D: DBAccess> {
+    db: Arc<D>,
+    // The Database needs to outlive the snapshot, and be created by the supplied `db`.
+    inner: *const ffi::rocksdb_snapshot_t,
+}
+
+#[allow(dead_code)]
+impl<D: DBAccess> SnapshotWithDBArc<D> {
+    /// Creates a new `SnapshotWithDBArc` of the database `db`.
+    pub fn new(db: Arc<D>) -> Self {
+        let snapshot = unsafe { db.create_snapshot() };
+        Self { db, inner: snapshot }
+    }
+
+    /// Returns the bytes associated with a key value with default read options.
+    pub fn get<K: AsRef<[u8]>>(&self, key: K) -> Result<Option<Vec<u8>>, Error> {
```

Comment:
How much clutter is it for the user of this API if we removed these overloads which don't have the `ReadOptions` specified, and the user calls the "opt" versions with `ReadOptions::default()`? It would tidy up the API here a bit, but then not sure if it's worth it.